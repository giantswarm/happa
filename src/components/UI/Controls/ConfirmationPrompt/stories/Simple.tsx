import { StoryFn } from '@storybook/react';
import { Box, Text } from 'grommet';
import React, { ComponentPropsWithoutRef, useState } from 'react';
import Button from 'UI/Controls/Button';

import ConfirmationPrompt from '..';

export const Simple: StoryFn<
  ComponentPropsWithoutRef<typeof ConfirmationPrompt>
> = (args) => {
  const [open, setIsOpen] = useState(false);

  return (
    <Box>
      <Box margin={{ bottom: 'small' }}>
        <Button onClick={() => setIsOpen(!open)}>Open prompt</Button>
      </Box>
      <ConfirmationPrompt {...args} open={open}>
        <Text>Check out this awesome component!</Text>
      </ConfirmationPrompt>
    </Box>
  );
};
