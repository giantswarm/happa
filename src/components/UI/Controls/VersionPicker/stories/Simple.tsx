import { action } from '@storybook/addon-actions';
import { Story } from '@storybook/react/types-6-0';
import React, { ComponentPropsWithoutRef } from 'react';

import VersionPicker from '../VersionPicker';

export const Simple: Story<ComponentPropsWithoutRef<typeof VersionPicker>> = (
  args
) => {
  return (
    <VersionPicker
      onChange={action('changed')}
      selectedVersion={args.selectedVersion}
      versions={[
        {
          created: '2020-12-15T16:39:27Z',
          includesVersion: '2.0.0',
          chartVersion: '1.0.5',
          test: false,
        },
        {
          created: '2020-12-14T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '1.0.3',
          test: false,
        },
        {
          created: '2020-12-13T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '1.0.2',
          test: false,
        },
        {
          created: '2020-12-12T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '0.8.9',
          test: false,
        },
        {
          created: '2020-12-11T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '0.8.8',
          test: false,
        },
        {
          created: '2020-12-10T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '0.8.7',
          test: false,
        },
        {
          created: '2020-11-10T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '0.8.6',
          test: false,
        },
        {
          created: '2020-10-10T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '0.8.5',
          test: false,
        },
        {
          created: '2020-09-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.8.4',
          test: false,
        },
        {
          created: '2020-08-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.8.3',
          test: false,
        },
        {
          created: '2020-07-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.8.2',
          test: false,
        },
        {
          created: '2020-06-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.8.1',
          test: false,
        },
        {
          created: '2020-05-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.8.0',
          test: false,
        },
        {
          created: '2020-04-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.7.10',
          test: false,
        },
        {
          created: '2020-03-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.7.8',
          test: false,
        },
        {
          created: '2020-02-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.7.3',
          test: false,
        },
      ]}
    />
  );
};

Simple.args = {
  selectedVersion: '1.0.2',
};

Simple.argTypes = {
  selectedVersion: {
    description: 'The version selected when rendering the component',
    control: { type: 'text' },
  },
};
