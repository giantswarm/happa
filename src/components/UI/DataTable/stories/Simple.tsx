import { Story } from '@storybook/react';
import { DataTableProps, Text } from 'grommet';
import React from 'react';

import DataTable from '../';

interface IData {
  name: string;
  location: string;
  amount: number;
}

export const Simple: Story<DataTableProps> = (args) => {
  return <DataTable<IData> {...args} />;
};

Simple.args = {
  columns: [
    {
      property: 'name',
      header: 'Name',
      primary: true,
    },
    {
      property: 'location',
      header: 'Location',
      sortable: false,
    },
    {
      property: 'amount',
      header: 'Amount',
      render: (data) => <Text>${data.amount}</Text>,
    },
  ],
  data: [
    {
      name: 'Food',
      location: 'Amazon',
      amount: 25.36,
    },
    {
      name: 'Laptop',
      location: 'Apple',
      amount: 1683.0,
    },
    {
      name: 'Phone',
      location: 'Apple',
      amount: 1037.73,
    },
  ],
  sortable: true,
};

Simple.argTypes = {};
