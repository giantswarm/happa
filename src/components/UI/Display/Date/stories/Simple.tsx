import { Story } from '@storybook/react';
import React from 'react';

import Date from '..';

export const Simple: Story<React.ComponentPropsWithoutRef<typeof Date>> = (
  args
) => {
  return (
    <div>
      <Date {...args} />
    </div>
  );
};

Simple.args = {};

Simple.argTypes = {
  value: {
    control: {
      type: 'date',
    },
    defaultValue: '2021-01-10',
    name: 'date (local time zone)',
  },
};
