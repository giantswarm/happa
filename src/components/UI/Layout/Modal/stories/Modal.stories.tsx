import { Meta } from '@storybook/react';

import Modal from '..';

export { Simple } from './Simple';
export { WithFooter } from './WithFooter';

export default {
  title: 'Layout/Modal',
  component: Modal,
} as Meta;
