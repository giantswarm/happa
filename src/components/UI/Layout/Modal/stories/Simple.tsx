import { Story } from '@storybook/react';
import { Text } from 'grommet';
import React, { ComponentPropsWithoutRef } from 'react';

import Modal from '..';

export const Simple: Story<ComponentPropsWithoutRef<typeof Modal>> = (args) => {
  return (
    <Modal {...args}>
      <Text>Welcome to my new modal!</Text>
    </Modal>
  );
};

Simple.args = {
  title: 'Hello friend',
  visible: true,
};

Simple.argTypes = {
  title: {
    description: 'The text to display at the top of the modal.',
    control: {
      type: 'text',
    },
  },
  visible: {
    description: 'Whether the modal should be visible or not.',
    control: {
      type: 'boolean',
    },
  },
};
