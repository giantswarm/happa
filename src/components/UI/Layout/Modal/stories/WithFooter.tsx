import { Story } from '@storybook/react';
import { Text } from 'grommet';
import React, { ComponentPropsWithoutRef } from 'react';

import Modal from '..';

export const WithFooter: Story<ComponentPropsWithoutRef<typeof Modal>> = (
  args
) => {
  return (
    <Modal {...args}>
      <Text>Welcome to my new modal!</Text>
    </Modal>
  );
};

WithFooter.args = {
  title: 'Hello friend',
  visible: true,
  footer: `I'm the footer around here`,
};

WithFooter.argTypes = {
  title: {
    description: 'The text to display at the top of the modal.',
    control: {
      type: 'text',
    },
  },
  visible: {
    description: 'Whether the modal should be visible or not.',
    control: {
      type: 'boolean',
    },
  },
};
