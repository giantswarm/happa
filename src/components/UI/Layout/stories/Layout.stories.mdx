import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';
import Well from '../Well';
import Aside from '../Aside';
import DetailList from '../DetailList';
import Section from '../Section';

<Meta title="Layout/Layout" />

# Layout

TODO: Add examples for `FlexDivs` and `TwoInputArea`.

## Aside

<Preview>
  <Story name="Aside">
    <p>A paragraph of text before.</p>
    <Aside>
        <p>Here is some content inside an aside.</p>
    </Aside>
    <p>A paragraph of text below.</p>
   </Story>
</Preview>

## DetailList

<Preview>
  <Story name="DetailList">
    <DetailList title="Title of the list">
        <p>Here is a first paragraph.</p>
        <p>Then comes another paragraph.</p>
    </DetailList>
   </Story>
</Preview>

## Section

<Preview>
  <Story name="Section">
    <Section title="The section title">
        <p>This is a paragraph of text inside our section. It should appear next to the section title.</p>
    </Section>
  </Story>
</Preview>

<Preview>
   <Story name="Section (flat)">
    <Section title="The section title" flat>
        <p>This is a paragraph of text inside our section. It should appear beneath the section title.</p>
    </Section>
   </Story>
</Preview>

## Well

<Preview>
  <Story name="Well">
    <p>A paragraph of text before.</p>
    <Well>
        <p>Here is a paragraph of text inside a well.</p>
    </Well>
    <p>A paragraph of text below.</p>
   </Story>
</Preview>
