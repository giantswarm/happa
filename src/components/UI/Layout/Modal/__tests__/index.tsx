import { renderWithTheme } from 'test/renderUtils';

import { Simple } from '../stories/Simple';
import { WithFooter } from '../stories/WithFooter';

describe('Modal', () => {
  it('renders a closed modal', () => {
    const { container } = renderWithTheme(Simple, {
      title: 'Hello friend',
      children: 'Welcome to my new modal!',
      visible: false,
      onClose: jest.fn(),
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders an opened modal', () => {
    const { container } = renderWithTheme(Simple, {
      title: 'Hello friend',
      children: 'Welcome to my new modal!',
      visible: true,
      onClose: jest.fn(),
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders a modal with a footer', () => {
    const { container } = renderWithTheme(WithFooter, {
      title: 'Hello friend',
      children: 'Welcome to my new modal!',
      visible: true,
      footer: `I'm the footer around here`,
      onClose: jest.fn(),
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
