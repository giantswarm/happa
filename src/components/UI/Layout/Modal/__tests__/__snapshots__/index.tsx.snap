// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders a closed modal 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 hRVLan"
>
  <div
    class="ThemeProvider__AppWrapper-sc-12xd2cf-0 eKUXDU"
  />
</div>
`;

exports[`Modal renders a modal with a footer 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 hRVLan"
>
  <div
    class="ThemeProvider__AppWrapper-sc-12xd2cf-0 eKUXDU"
  />
</div>
`;

exports[`Modal renders an opened modal 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 hRVLan"
>
  <div
    class="ThemeProvider__AppWrapper-sc-12xd2cf-0 eKUXDU"
  />
</div>
`;
